xapian-letor/tests/api_letor.cc

```diff
@@ -263,6 +263,36 @@ DEFINE_TESTCASE(preparetrainingfileonedb, generated)
     return true;
 }
 
+DEFINE_TESTCASE(test_parse_querystring, generated)
+{
+    // All those cases which are not valid.
+    string db_path = get_database_path("apitest_listnet_ranker1",
+				       db_index_one_document);
+    string data_directory = test_driver::get_srcdir() + "/testdata/";
+    string query = data_directory + "parse_query_noopenquote.txt";
+    string qrel = data_directory + "qrelone.txt";
+    string training_data = data_directory + "training_output.txt";
+    TEST_EXCEPTION(Xapian::LetorParseError,
+		   Xapian::prepare_training_file(db_path, query, qrel, 10,
+						 "training_output.txt"));
```

Comment:
With three cases (and especially if you add more) I might be inclined to define a `TEST_PARSE_EXCEPTION` macro so each instance becomes a single line. (You could also inline the `query` expressions, since they change for every test.)

Reply:
I meant go even further, because the invocations are so similar:

```c++
#define TEST_PARSE_EXCEPTION(testfile) TEST_EXCEPTION(Xapian::LetorParseError,\
    Xapian::prepare_training_file(db_path, data_directory + testfile, qrel, 10, "training_output.txt"));
```

or something like that.

Reply:
Ah I will change that now

Reply:
fixed in a fixup commit.